modules/external-api/src/main/java/org/opencastproject/external/endpoint/EventsEndpoint.java

```diff
@@ -421,14 +433,45 @@ public Response getEventMedia(@HeaderParam("Accept") String acceptHeader, @PathP
   @Path("{eventId}")
   @RestQuery(name = "deleteevent", description = "Deletes an event.", returnDescription = "", pathParameters = {
           @RestParameter(name = "eventId", description = "The event id", isRequired = true, type = STRING) }, responses = {
-                  @RestResponse(description = "The event has been deleted.", responseCode = HttpServletResponse.SC_NO_CONTENT),
-                  @RestResponse(description = "The specified event does not exist.", responseCode = HttpServletResponse.SC_NOT_FOUND) })
+          @RestResponse(description = "The event has been deleted.", responseCode = HttpServletResponse.SC_NO_CONTENT),
+          @RestResponse(description = "The retraction of publications has started.", responseCode = HttpServletResponse.SC_ACCEPTED),
+          @RestResponse(description = "The specified event does not exist.", responseCode = HttpServletResponse.SC_NOT_FOUND) })
   public Response deleteEvent(@HeaderParam("Accept") String acceptHeader, @PathParam("eventId") String id)
-          throws NotFoundException, UnauthorizedException {
-    if (!indexService.removeEvent(id))
-      return Response.serverError().build();
-
-    return Response.noContent().build();
+          throws SearchIndexException, UnauthorizedException {
+    final Opt<Event> event = indexService.getEvent(id, externalIndex);
+    if (event.isNone()) {
+      return RestUtil.R.notFound(id);
+    }
+    final Runnable doOnNotFound = () -> {
+      try {
+        externalIndex.delete(Event.DOCUMENT_TYPE,id.concat(getSecurityService().getOrganization().getId()));
```

Comment:
This line no longer works with the index changes -> incorrect number of arguments.

Reply:
Sorryyyy

Reply:
Ok, rebased to `r/10.x` and changed the `delete` method arguments ... maybe just one time this gets merged before Katrin does the next change ;-)

Reply:
Sounds like a challenge ;)